app/services/report.py

```diff
@@ -29,7 +29,10 @@ def add_report(self, twitter_id: str, reporter_id: str) -> Report:
         """
         report_db = Report.objects(twitter_id=twitter_id).first()
         if report_db is None:
-            user = self.twitter_scraper.get_user_by_id(twitter_id)
+            user = TwitterUser.objects(twitter_id=twitter_id).first()
+
+            if user is None:
+                raise HTTPException(404, "Twitter account not in database")
```

Comment:
Note that this is a new flow. We should discuss abt this. we can either:
1. Throw error TwitterAccount does not exist and frontend tell user to re-check before reporting
2. Not throw error and go scrape+check the user ourselves if the old data is already deleted 

**`Maybe we should go with the 1st option`** since check result could change.

Reply:
This error cannot happen from **frontend**. It can only happen if you send request directly to backend.

Reply:
Can you elaborate this?

Reply:
You can report on homepage and detail page, and to display these pages you need to fetch Twitter user information anyway, so it is already in database.

Reply:
And the case where `TwitterUser` gets deleted by background job?

Reply:
When you go to those page, it will fetch Twitter user information again anyway.

Reply:
Not sure about that, there's a gap between going to `detail`/`home` page and when user actually `report`.

Reply:
What if when we fetch missing user information and run ML model, the new check result is different from old result displayed on frontend?

Reply:
That's why i gave 2 approaches

Reply:
Ok then go with the 1st option